docs/src/index.md

```diff
@@ -1,37 +1,41 @@
 # DensityInterface.jl
 
-This package defines an interface for mathematical/statistical densities in Julia. The interface comprises the functions [`isdensitytype`](@ref),  [`logdensityof`](@ref) and [`logfuncdensity`](@ref).
+```@docs
+DensityInterface
+```
+
+This package defines an interface for mathematical/statistical densities and objects associated with a density in Julia. The interface comprises the functions [`hasdensity`](@ref),  [`logdensityof`](@ref)/[`densityof`](@ref) and [`logfuncdensity`](@ref).
 
-A the following methods must be provided to make a type (e.g. `SomeDensity`) compatible with the interface:
+The following methods must be provided to make a type (e.g. `SomeDensity`) compatible with the interface:
 
 ```julia
 import DensityInterface
 
-DensityInterface.isdensitytype(::Type{<:SomeType}) = true
-DensityInterface.logdensityof(density::SomeDensity, x) = log_of_density_at_x
+DensityInterface.hasdensity(::SomeDensity) = true
+DensityInterface.logdensityof(d::SomeDensity, x) = log_of_d_at_x
 ```
 
-DensityInterface includes a default implementation of `logdensityof(density)`. It provides a convenient way of passing a log-density function to algorithms like optimizers, samplers, etc.:
+The object `d` may be a density itself or something that can be said to have a density. If `d` is a distribution, the density is the PDF. If `d` is a measure in general, it's density is implied here to be the Radon–Nikodym derivative of `d` and it's base measure. In statistical inference applications, for example, `d` might be a likelihood, prior or posterior.
 
-```julia
-using DensityInterface
+Note: The package [`MeasureTheory`](https://github.com/cscherrer/MeasureTheory.jl) provides tools to work with densities and measures that go beyond the density in respect to an implied base measure.
```

Comment:
Maybe this could be added to a potential footnote as well to disturb the reading flow a bit less and to allow readers ro skip the measure theory parts more easily. Alternatively, everything could be moved to a separate section such as "Measure theoretical view".

Reply:
Done.